controllers/storageautoscaler/verify_test.go

```diff
@@ -25,3 +30,30 @@ func TestTimeoutHasElapsed(t *testing.T) {
 	fail = timeoutHasElapsed(timeoutSeconds, startTime)
 	assert.True(t, fail)
 }
+
+func TestVerifyVerticalScaling(t *testing.T) {
+	// mock getOsdSize() function for test
+	mockGetOsdSize = func(ctx context.Context, namespace string, log logr.Logger) (model.Vector, error) {
+		return model.Vector{
+			&model.Sample{
+				Metric: model.Metric{
+					"device_class": "hdd",
+				},
+				Value: model.SampleValue(10),
+			},
+		}, nil
```

Comment:
Verifying that scaling has/hasn't worked is one of the core pieces of logic that this feature depends on. Do you think that a single unit test that covers only the basic happy path is sufficient to prevent regressions?

Reply:
Still need to add more test cases to it, for sure this is the crucial part of the logic, waiting on the new design to finalsied so we can act on this accordingly.
https://github.com/red-hat-storage/ocs-operator/pull/3105